src/api/app/helpers/maintenance_helper.rb

```diff
@@ -190,29 +193,37 @@ def copy_binaries(filter_source_repository, source_package, target_package_name,
     update_ids
   end
 
-  def copy_binaries_to_repository(source_repository, source_package, target_repo, target_package_name, setrelease)
+  def copy_binaries_to_repository(source_repository, source_package, target_repo, target_package_name,
+                                  multibuild_container, setrelease)
     u_id = get_updateinfo_id(source_package, target_repo)
+    source_package_name = source_package.name
+    unless multibuild_container.blank?
+      source_package_name << ":" << multibuild_container
+      target_package_name = target_package_name.gsub(/:.*/, '') << ":" << multibuild_container
+    end
     source_repository.architectures.each do |arch|
       # get updateinfo id in case the source package comes from a maintenance project
-      copy_single_binary(arch, target_repo, source_package, source_repository, target_package_name, u_id, setrelease)
+      copy_single_binary(arch, target_repo, source_package.project.name, source_package_name,
+                         source_repository, target_package_name, u_id, setrelease)
     end
     u_id
   end
 
-  def copy_single_binary(arch, target_repository, source_package, source_repo, target_package_name, update_info_id, setrelease)
+  def copy_single_binary(arch, target_repository, source_project_name, source_package_name, source_repo,
+                         target_package_name, update_info_id, setrelease)
     cp_params = {
       cmd:         "copy",
-      oproject:    source_package.project.name,
-      opackage:    source_package.name,
+      oproject:    source_project_name,
+      opackage:    source_package_name,
       orepository: source_repo.name,
       user:        User.current.login,
-      multibuild:  "1",
       resign:      "1"
     }
     cp_params[:setupdateinfoid] = update_info_id if update_info_id
     cp_params[:setrelease] = setrelease if setrelease
+    cp_params[:multibuild] = "1" unless source_package_name.include? ':'
     # rubocop:disable Metrics/LineLength
-    cp_path = "/build/#{CGI.escape(target_repository.project.name)}/#{URI.escape(target_repository.name)}/#{URI.escape(arch.name)}/#{URI.escape(target_package_name)}"
+    cp_path = "/build/#{URI.escape(target_repository.project.name)}/#{URI.escape(target_repository.name)}/#{URI.escape(arch.name)}/#{URI.escape(target_package_name)}"
```

Comment:
Don't use `URI.escape`, is going to be deprecated and doesn't really work well for urls... so, do it the other way around, switch `URI.escape` calls to `CGI.escape`.

https://github.com/alexreisner/geocoder/issues/609

I'm trying to change those in all the url calculations for the backend at least in the Backend::Connection calls

Reply:
I remember that we had to fix encoding bugs, by replacing CGI.escape with URI.escape.

Eg. a space is encoded different, so how can this be obsoleted?

Reply:
There is a nice summary on stackoverflow about URI.escape, CGI.escape and Co. You find it in the third answer [here](https://stackoverflow.com/questions/2824126/whats-the-difference-between-uri-escape-and-cgi-escape).

@adrianschroeter Might be that in the cases you mention you actually want to use addressable (see the second part of the stackoverflow answer).

Reply:
okay, we need to find out what the RFC's really say ... but the backend is only decoding %XX encodings in path atm so this would break (if we allow somewhere a space at all in the path, but IIRC we did somewhere).
So we either need to change the backend or keep using URI.escape in the api...

Reply:
btw, we allow package names with a "+" inside. this would conflict with this encoding, I suppose?
We really need some more testing if we really want to change this at least ...

Reply:
From the second answer of the link bgeuken commented before.

> The problem with URI.escape was that it could not handle the RFC-3896 spec.

And the conclusions there are:

>Conclusion:
>
>  *  Do not use URI.escape or similar
>  *   Use CGI::escape if you only need form escape
>  *  If you need to work with URIs, use Addressable, it offers URL encoding, form encoding and normalizes URLs.
>  *  If it is a Rails project, check out "How do I URL-escape a string in Rails?"

So, we need to dig into what the backend is compliant and rework those access to the backend in concordance.

Reply:
Regarding the CGI.escape switch, Michael and me looked at it it should work in most cases, but we found one case where it will break: We allow currently spaces as repository names and you can reach them via path routes. Encoding a space there to a + would break there to a not found repository.

Maybe the suggessted URI.escape_www_form or alike is a better take here?

In the end the name from CGI nor the URI. form_encode methods really match the use case.

We may also want to disallow repositories with spaces in name, but this would be an incompatible change which breaks builds and download URLs.

Reply:
> We may also want to disallow repositories with spaces in name, but this would be an incompatible change which breaks builds and download URLs.

What  about changing the backend to be url compliant like it should be? 
I mean, if the backend decodes url data following the RFC for http, then the + symbols to spaces won't be a problem then.

Reply:
Ping @adrianschroeter @mlschroe .... what about this discussion??